python/lsst/cp/pipe/ptc/cpExtractPtcTask.py

```diff
@@ -313,14 +349,29 @@ def run(self, inputExp, inputDims, taskMetadata):
         # dimensions match.
         dummyPtcDataset = PhotonTransferCurveDataset(ampNames, 'DUMMY',
                                                      self.config.maximumRangeCovariancesAstier)
-
-        readNoiseDict = {ampName: 0.0 for ampName in ampNames}
         for ampName in ampNames:
-            # Initialize amps of `dummyPtcDatset`.
             dummyPtcDataset.setAmpValues(ampName)
-            # Overscan readnoise from post-ISR exposure metadata.
-            # It will be used to estimate the gain from a pair of flats.
-            readNoiseDict[ampName] = self.getReadNoiseFromMetadata(taskMetadata, ampName)
+        # Get read noise.  Try from the exposure, then try
+        # taskMetadata.  This adds a get() for the exposures.
+        readNoiseLists = {}
+        for pairIndex, expRefs in inputExp.items():
+            # This yields an index (exposure_time, seq_num, or flux)
+            # and a pair of references at that index.
+            for expRef, expId in expRefs:
+                # This yields an exposure ref and an exposureId.
+                exposure = expRef.get()
```

Comment:
Can't we just get the metadata component rather than the full exposure?  Or is there something else we need?

Reply:
I simply didn't think of this at the time.  Switched to get just the metadata.